src/background/streams.ts

```diff
@@ -51,11 +52,14 @@ export class PortStreams extends Duplex {
 
   _write(chunk: any, encoding: string, callback: (error?: Error | null) => void): void {
     if (isRuntimeMessage(chunk)) {
-      const handle = (dst: string) => {
+      const handle = (dst: string, type) => {
         if (dst === MessageOrigin.BG) {
-          this.push(chunk);
+          this.push(chunk);// TODO: portStream.pipe(this); _write this.push(chunk)
         } else {
-          const streams = this.streams.filter(stream => stream.port.name.startsWith(dst));
+          const streams = this.streams.filter(
+            stream => stream.port.name.startsWith(dst) &&
+              (this.enabledPortNames.includes(stream.port.name) || type === BgMsgTypes.ENABLE_RESPONSE)
```

Comment:
should allow communication between BG and popup

Reply:
Ah, you are right. BG will need to communicate with popup even `enable()` is not called. Will change this